### PR TITLE
[WIP] geometry.depth submodule

### PR DESCRIPTION
**Pull request by edgarriba**
Pull request that implements few fiunctionalities for rgbd:
- [x] `depth_to_3d`
- [x] `depth_to_normals`
- [x] `warp_frame_depth`

TODO:
- [x] add numeric tests

### CODE REVIEW

File: kornia/geometry/depth.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        raise ValueError(f"Input src_trans_dst must have a shape (B, 3, 3). "
+                         f"Got: {src_trans_dst.shape}.")
+
+    if not isinstance(camera_matrix, torch.Tensor):
+        raise TypeError(f"Input camera_matrix type is not a torch.Tensor. "
+                        f"Got {type(camera_matrix)}.")
+
+    if not len(camera_matrix.shape) == 3 and camera_matrix.shape[-2:] == (3, 3):
+        raise ValueError(f"Input camera_matrix must have a shape (B, 3, 3). "
+                         f"Got: {camera_matrix.shape}.")
```
Comment (ducha-aiki) [inline kornia/geometry/depth.py]: May be for the next PRs we could refactor out checks or messages to aux functions?

File: kornia/geometry/depth.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    # unproject source points to camera frame
+    points_3d_dst: torch.Tensor = depth_to_3d(depth_dst, camera_matrix)  # Bx3xHxW
+
+    # transform points from source to destionation
+    points_3d_dst = points_3d_dst.permute(0, 2, 3, 1)  # BxHxWx3
+
+    # hack to match sizes
+    batch_size: int = image_src.shape[0]
```
Comment (ducha-aiki) [inline kornia/geometry/depth.py]: I though that pytorch convention is to use .size(0) over .shape[0].

File: test/geometry/test_depth.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    def test_gradcheck(self):
+        # generate input data
+        depth = torch.rand(1, 1, 3, 4)
+        depth = utils.tensor_to_gradcheck_var(depth)  # to var
+
+        camera_matrix = torch.rand(1, 3, 3)
+        camera_matrix = utils.tensor_to_gradcheck_var(camera_matrix)  # to var
+
+        # evaluate function gradient
+        assert gradcheck(kornia.depth_to_3d, (depth, camera_matrix,),
```
Comment (ducha-aiki) [inline test/geometry/test_depth.py]: Should we also add toy-example test with numbers? Like this. 
https://github.com/arraiyopensource/kornia/blob/master/test/filters/test_gaussian.py#L78

Same for the rest of test. We don`t have such for all existing code, but I think, to have one is a good thing.

Comment (edgarriba) [reply test/geometry/test_depth.py]: sure, will add it in the next commit

File: kornia/geometry/depth.py (hunk 1 above)
Comment (edgarriba) [reply kornia/geometry/depth.py]: totally agree, we have to write all the time same exceptions

File: kornia/geometry/depth.py (hunk 2 above)
Comment (edgarriba) [reply kornia/geometry/depth.py]: not sure if it's a convention, I use .shape to make it more numpy like. We can decide on that.

File: kornia/geometry/depth.py (hunk 1 above)
Comment (ducha-aiki) [reply kornia/geometry/depth.py]: what about this format?
https://github.com/ducha-aiki/kornia/blob/rebased-local-feature-detection/kornia/feature/laf.py#L8

**Pull request merged**